syft/frameworks/torch/tensors/decorators/sensitivity.py

```diff
@@ -0,0 +1,446 @@
+    def __gt__(self, other):
+        """BUG: the zero values mess this up"""
+        if isinstance(other, SensitivityTensor):
+
+            new_vals = self.values > other.values
+
+            # if self is bigger than the biggest possible other
+            if_left = (self.min_vals > other.expanded_maxmaxvals).float() * self.entities
+
+            # if self is smaller than the smallest possible other
+            if_right = (self.max_vals < other.expanded_minminvals).float() * self.entities
+
+            # if self doesn't overlap with other at all
+            if_left_or_right = if_left + if_right  # shouldn't have to check if this > 2 assuming
+            # other's max is > other's min
+
+            # if self does overlap with other
+            new_self_max_vals = 1 - if_left_or_right
+
+            # can't have a threshold output less than 0
+            new_self_min_vals = if_left_or_right * 0
+
+            # if other is bigger than the smallest possible self
+            if_left = (other.min_vals > self.expanded_maxmaxvals).float() * other.entities
+
+            # if other is smaller than the smallest possible self
+            if_right = (other.max_vals < self.expanded_minminvals).float() * other.entities
+
+            # if other and self don't overlap
+            if_left_or_right = if_left + if_right  # shouldn't have to check if this > 2 assuming
+            # other's max is > other's min
+
+            # if other and self do overlap
+            new_other_max_vals = 1 - if_left_or_right
+
+            # the smallest possible result is 0
+            new_other_min_vals = new_self_min_vals + 0
+
+            # only contribute information from entities in ancestry
+            new_self_max_vals = (new_self_max_vals * self.entities) + (
+                (1 - self.entities) * -max_long
+            )
+            new_other_max_vals = (new_other_max_vals * self.entities) + (
+                (1 - self.entities) * -max_long
+            )
+
+            # only contribute information from entities in ancestry
+            new_self_min_vals = (new_self_min_vals * self.entities) + (
+                (1 - self.entities) * max_long
+            )
+            new_other_min_vals = (new_other_min_vals * self.entities) + (
+                (1 - self.entities) * max_long
+            )
+
+            entities_self_or_other = ((self.entities + other.entities) > 0).float()
+
+            new_max_val = th.max(new_self_max_vals, new_other_max_vals) * entities_self_or_other
+            new_min_val = th.min(new_self_min_vals, new_other_min_vals) * entities_self_or_other
+
+        else:
+
+            new_vals = self.values > other
+
+            if_left = other <= self.max_vals
+            if_right = other >= self.min_vals
+            if_and = if_left * if_right
+
+            new_max_val = if_and
+            new_min_val = new_max_val * 0
+
+        return SensitivityTensor(new_vals, new_max_val, new_min_val)
+
+    def __lt__(self, other):
```

Comment:
Can we define less than in terms of gt or vice versa to better reuse code?

Reply:
erm - maybe - i'll have to get back to you on that.

Reply:
we can definitely get at least come code reuse there - maybe break out the internals into a helper method or two

Reply:
the min/max manipulation is very similar between the two

Reply:
I'm pretty sure gt is just lt with self and other switched

Reply:
oh... fair point

Reply:
done